app/controllers/development_metrics_controller.rb

```diff
@@ -66,6 +77,48 @@ def build_metrics_definitions
     @pull_request_size_definition = MetricDefinition.find_by(code: :pull_request_size)
   end
 
+  def set_metrics_to_show
+    metrics_to_show if @query_metric_name.present?
+
+    show_defect_escape_rate
+    show_development_cycle
+  end
+
+  def metrics_to_show
+    case @query_metric_name.to_sym
+    when :defect_escape_rate
+      @show_defect_escape_rate = true
+    when :development_cycle
+      @show_development_cycle = true
+    end
+  end
+
+  def defect_escape_rate(metrics)
+    return unless @show_defect_escape_rate
+
+    @defect_escape_rate = metrics[:defect_escape_rate]
+    @metric_definition = MetricDefinition.find_by(code: :defect_escape_rate)
```

Comment:
` @metric_definition` this name does not seem very descriptive since, as far as I understand, it refers only to defect escape rate definition. For example you used `@development_cycle_definition` below, that is more accurate. 

if it involves a lot of change don't worry. Just personal taste here.

Reply:
Done! I haven't notice but is better for readability.